File: maxhanna.Server/Controllers/DataContracts/ArrayCharacter.cs
namespace maxhanna.Server.Controllers.DataContracts
{
    public class ArrayCharacter
    {
        public User User { get; set; }
        public int CharacterClass { get; set; }
        public long Level { get; set; }
        public long Experience { get; set; }
        public long Position { get; set; }
        public long MonstersKilled { get; set; }
        public int PlayersKilled { get; set; }
        public long ItemsFound { get; set; }

        public ArrayCharacter(User user, int characterClass, long level, long experience, long position, long monstersKilled, int playersKilled, long itemsFound)
        {
            User = user;
            CharacterClass = characterClass;
            Level = level;
            Experience = experience;
            Position = position;
            MonstersKilled = monstersKilled;
            PlayersKilled = playersKilled;
            ItemsFound = itemsFound;
        }
        public ArrayCharacter(User user)
        {
            User = user;
        }
    }
}

File: maxhanna.client/src/services/datacontracts/array/array-character.ts
import { User } from "../user/user";

export class ArrayCharacter {
  characterClass: number | undefined;
  level: number | undefined;
  experience: number | undefined;
  position: number | undefined;
  monstersKilled: number | undefined;
  itemsFound: number | undefined;
  playersKilled: number | undefined;
  user: User | undefined;

  constructor( 
    characterClass: number = 0,
    level: number = 0,
    experience: number = 0,
    position: number = 0,
    monstersKilled: number = 0,
    itemsFound: number = 0,
    playersKilled: number = 0,
    user?: User
  ) { 
    this.characterClass = characterClass;
    this.level = level;
    this.experience = experience;
    this.position = position;
    this.monstersKilled = monstersKilled;
    this.itemsFound = itemsFound;
    this.playersKilled = playersKilled;
    this.user = user;
  }
}
